python/src/scipp/typing.py

```diff
@@ -43,9 +43,9 @@ def has_numeric_type(obj: _std_typing.Any) -> bool:
     return (not has_vector_type(obj)) and (not has_string_type(obj))
 
 
-#: Any object that behaves like a scipp.DataArray in most operations.
-#: This explicitly excludes Datasets.
-DataArrayLike = _std_typing.Union[sc.Variable, sc.DataArray]
+#: Any object that behaves like a scipp.Variable,
+#:  that is an array with labeled dimensions.
+LabeledArray = _std_typing.Union[sc.Variable, sc.DataArray, sc.Dataset]
```

Comment:
Oh, now I understand, "labeled" refers to the dims, not the coords (which would be labeling the elements). Then the name does make sense, but is still ambiguous. :(

Reply:
Suggestions are welcome!

We could just use `VariableLike`?

Reply:
No idea, I am giving up for now, I think :(